### PR TITLE
Step 4 POST

### PR DESCRIPTION
**Pull request by andyp21**
- create a new Java Servlet file DataServlet_rev.java
- implement form POST which reverses the text input

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet_rev.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/reverse")
+public class DataServlet_rev extends HttpServlet {
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet_rev.java:26]: DataServletReverse would be a better name, underscores are not used for Java naming conventions. Same for the StringBuilder

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet_rev.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    
+    // Reverses String text
+    StringBuilder text_rev = new StringBuilder();
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet_rev.java:34]: You can directly initialize the StringBuilder to the text entered, there is a constructor for that

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -70,13 +70,23 @@ <h2 style= "margin: 0 5% 0 5%">Princeton University</h2>
...
                     <h1>Fetch</h1>
                     <p>Click here to see Favourite Ice-creams:</p>
                     <button onclick="getJson()">Fetch statement!</button>
                     <div id="gen-container"></div>
 
                 <!-- Form -->
+                <form action="/reverse" method="POST">
+
+                    <p>Enter a word to see its reverse:</p>
+                    <textarea name="text-input">teem</textarea>
```
Comment (ifranch) [inline portfolio/src/main/webapp/index.html:83]: Is this a typo? You don't need to have any text in the <textarea>

Comment (andyp21) [reply portfolio/src/main/webapp/index.html:83]: It was an example of the types of words people could perhaps try entering. 'teem' would be flipped to 'meet'

Comment (ifranch) [reply portfolio/src/main/webapp/index.html:83]: Ooh ok makes sense

**Pull request merged**